packages/models/src/crypto-assets/crypto-asset-balance.model.ts

```diff
@@ -0,0 +1,49 @@
+import { MarketData } from '../market.model';
+import { Money } from '../money.model';
+
+/** Balance types:
+ * availableBalance: totalBalance after filtering out outboundBalance, protectedBalance, and uneconomicalBalance
+ * availableUnlockedBalance; availableBalance minus lockedBalance
+ * inboundBalance - balance of pending receipt into account given pending transactions
+ * lockedBalance - totalBalance minus total amount locked by contracts
+ * outboundBalance - balance of pending delivery from account given pending transactions
+ * pendingBalance - totalBalance plus inboundBalance minus outboundBalance
+ * protectedBalance - balance of UTXOs with collectibles (if bitcoin)
+ * totalBalance - balance as confirmed on chain
+ * uneconomicalBalance - balance across UTXOs with need for larger fee than principal by UTXO given standard rate (if bitcoin)
+ * unlockedBalance - totalBalance minus lockedBalance
+ */
+export interface Balance {
+  inboundBalance: Money;
+  outboundBalance: Money;
+  pendingBalance: Money;
+  totalBalance: Money;
+}
+
+export interface BtcBalance extends Balance {
+  protectedBalance: Money;
+}
+
+export interface StxBalance extends Balance {
+  availableUnlockedBalance: Money;
+  lockedBalance: Money;
+  unlockedBalance: Money;
+}
+
+export interface CryptoAssetBalance {
+  readonly availableBalance: Money;
+  readonly marketData: MarketData;
+}
+
+export interface BtcCryptoAssetBalance extends CryptoAssetBalance {
+  readonly balance: BtcBalance;
+}
+
+export interface StxCryptoAssetBalance extends CryptoAssetBalance {
+  readonly balance: StxBalance;
+}
```

Comment:
This creates a kind of nesting that seems odd to me. 

Separately, as before, in `CryptoAssetBalance` we're pairing models for a user's balance, and market data together. What's the rationale for creating this paring?

Reply:
I keep getting confused on your feedback, I swear you asked me to make sure the price data was available and accessible thru the data model when we query it? So, we always `select` the query and make the model with the price data in a hook?

Reply:
Will refactor and separate.